packages/agent/src/display.ts

```diff
@@ -194,30 +276,60 @@ export function getCredentialExchangeForDisplay(
   }
 }
 
-export function getW3cCredentialForDisplay(w3cCredentialRecord: W3cCredentialRecord) {
-  const credential = JsonTransformer.toJSON(
-    w3cCredentialRecord.credential.claimFormat === ClaimFormat.JwtVc
-      ? w3cCredentialRecord.credential.credential
-      : w3cCredentialRecord.credential
-  ) as W3cCredentialJson
-
-  const openId4VcMetadata = getOpenId4VcCredentialMetadata(w3cCredentialRecord)
-  const issuerDisplay = getIssuerDisplay(credential, openId4VcMetadata)
-  const credentialDisplay = getW3cCredentialDisplay(credential, openId4VcMetadata)
-
-  // FIXME: support credential with multiple subjects
-  const credentialAttributes = Array.isArray(credential.credentialSubject)
-    ? credential.credentialSubject[0] ?? {}
-    : credential.credentialSubject
-
-  return {
-    id: `w3c-credential-${w3cCredentialRecord.id}` satisfies CredentialForDisplayId,
-    createdAt: w3cCredentialRecord.createdAt,
-    display: {
-      ...credentialDisplay,
-      issuer: issuerDisplay,
-    },
-    w3cCredential: credential,
-    attributes: credentialAttributes,
+export function getCredentialForDisplay(credentialRecord: W3cCredentialRecord | SdJwtVcRecord) {
+  if (credentialRecord instanceof SdJwtVcRecord) {
+    // FIXME: we should probably add a decode method on the SdJwtVcRecord
+    // as you now need the agent context to decode the sd-jwt vc, while that's
+    // not really needed
+    const { decodedPayload } = decodeSdJwtVc(credentialRecord.compactSdJwtVc, (data, alg) =>
+      Hasher.hash(data, alg)
+    )
+
+    const openId4VcMetadata = getOpenId4VcCredentialMetadata(credentialRecord)
+    const issuerDisplay = getSdJwtIssuerDisplay(openId4VcMetadata)
+    const credentialDisplay = getSdJwtCredentialDisplay(decodedPayload, openId4VcMetadata)
+
+    // TODO: should we add a way to see these claims? Currently matching what would
+    // be included in a w3c credential subject
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { _sd_alg, _sd_hash, iss, vct, cnf, ...visibleProperties } = decodedPayload
+
+    // TODO: display somehow which fields can be selective disclosed
```

Comment:
Do we track this in code, or should we add it to a sprint board somewhere? I was testing with the wallet the other day and also saw that predicates for didcomm presentation requests are also not correct. Might be good to fix this sometime.

Reply:
Ah that predicates are not correct should not happen. Can you describe what the exact issue is?

I think we should start tracking these as tickets as well yes. I'll do a pass soon to get all these tickets in

Reply:
It's this FIXME: https://github.com/animo/paradym-wallet/blob/3f7e684796a6e6a8ca033fae681f7f83fcdd55d9/packages/agent/src/hooks/useAcceptDidCommPresentation.ts#L102